system/utils.py

```diff
@@ -533,13 +553,101 @@ def check_ledger_sync():
     demote_data = json.dumps({'alias': alias, 'services': []})
     demote_req = await ledger.build_node_request(trustee_did, target_did, demote_data)
     demote_res = json.loads(await ledger.sign_and_submit_request(pool_handle, wallet_handle, trustee_did, demote_req))
+    while demote_res['op'] != 'REPLY':
+        demote_res = json.loads(
+            await ledger.sign_and_submit_request(pool_handle, wallet_handle, trustee_did, demote_req))
     assert demote_res['op'] == 'REPLY'
 
 
 async def promote_node(pool_handle, wallet_handle, trustee_did, alias, target_did):
     promote_data = json.dumps({'alias': alias, 'services': ['VALIDATOR']})
     promote_req = await ledger.build_node_request(trustee_did, target_did, promote_data)
     promote_res = json.loads(await ledger.sign_and_submit_request(pool_handle, wallet_handle, trustee_did, promote_req))
+    while promote_res['op'] != 'REPLY':
+        promote_res = json.loads(
+            await ledger.sign_and_submit_request(pool_handle, wallet_handle, trustee_did, promote_req))
     host = testinfra.get_host('ssh://node'+alias[4:])
     host.run('systemctl restart indy-node')
     assert promote_res['op'] == 'REPLY'
+
+
+async def eventually_positive(func, *args, is_reading=False, is_self_asserted=False, cycles_limit=20):
+    cycles = 0
+    res = None
+
+    if is_self_asserted:  # this is for check_ledger_sync(), promote_node() and other self-asserted functions
```

Comment:
This looks like 3 totally different functions clumped into one. Why not just write separate `eventually_positive`, `write_eventually_positive` and `read_eventually_positive`?

Reply:
Done.